### PR TITLE
[server] Don't attribute usage to a team without active billing

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

Don't attribute usage to a team without active billing.

## Related Issue(s)

Fixes [issue in Slack](https://gitpod.slack.com/archives/C02EN94AEPL/p1666099912056929?thread_ts=1666083078.313979&cid=C02EN94AEPL) (internal)

## How to test

1. Create a new team called "Gitpod [Something]", but don't upgrade it for now
2. Add a new project under that team
3. Start a workspace for that project
4. Go to your personal billing settings (`/billing`) -- you should **not** be able to set the above team as default billing account
5. Upgrade the team to Usage-Based billing
6. In your personal billing settings (`/billing`) set your team as default billing account
7. Cancel the team's Usage-Based subscription again
8. Start a workspace for any repository that is not a project -- this should attempt to use your default billing account, but fail, and ask you to select a new billing account
9. Once the workspace is started, head over to your team's Usage view -- if you followed the above steps exactly, there should be 0 usage in this view

~~EDIT: Unfortunately, steps 7-9 cannot be tested in preview environments, because Stripe Webhooks are not supported in preview environments. I propose that we merge this PR, then test the final steps once deployed in production, and revert the PR if they don't work.~~

EDIT 2: Managed to configure a Stripe Webhook for this PR's preview environment! 🎉 Thanks @easyCZ and @andrew-farries for the help 🙌 

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-payment
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by jankeromnes** [comment]
Just ran through my test plan, and this failed 😳

> 8. Start a workspace for any repository that is not a project -- this should attempt to use your default billing account, but fail, and ask you to select a new billing account

It actually starts a workspace and successfully attributes usage to the team without billing 🤦 what is the validation code doing?

File: components/server/src/user/user-service.ts [hunk 1]
```diff
@@ -209,6 +216,12 @@ export class UserService {
                     "You're no longer a member of the selected billing team.",
                 );
             }
+            if (await this.isUnbilledTeam(attribution)) {
```
Comment (easyCZ) [inline components/server/src/user/user-service.ts:219]: 👍  This also mitigates https://github.com/gitpod-io/gitpod/issues/13977, but we should still prevent setting it in the first place (server-side).

File: components/server/src/user/user-service.ts [hunk 2]
```diff
@@ -236,7 +249,11 @@ export class UserService {
             } else {
                 attributionId = AttributionId.create(user);
             }
-            if (!!attributionId && (await this.hasCredits(attributionId))) {
+            if (
+                !!attributionId &&
+                (await this.hasCredits(attributionId)) &&
+                !(await this.isUnbilledTeam(attributionId))
+            ) {
```
Comment (easyCZ) [inline components/server/src/user/user-service.ts:256]: I found this quite complex to mentally parse, would it help if we wrote it as this?
```suggestion
            if (!!attributionId) {
              const enoughCredits = await this.hasCredits(attributionId);
              const teamWithoutBilling = await this.isUnbilledTeam(attributionId)
              if (enoughCredits && !teamWithoutBilling) {
                ...
              }
            }
            
```

**Review by easyCZ — APPROVED** [review]
/hold for Q on readability

**Comment by easyCZ** [comment]
Requires a rebase. Feel free to unhold.

**Pull request merged**